### PR TITLE
Refactor: Use generateUsername helper function

### PR DESCRIPTION
**Pull request by Vedant005**
`/packages/server/tests/utils/seed/user.ts` does not uniquely creates username.

This PR fixes it by using the generateUniqueUsername function to create new unique username

## Summary by CodeRabbit

* **Improvements**
  * Improved username handling: email-derived names are sanitized and truncated, a unique username is generated automatically when none is provided, and user-supplied usernames are still accepted.

* **Tests**
  * Added comprehensive unit tests for the username-generation utility covering trimming rules, separator behavior, suffix length, randomness, and edge cases.

### CODE REVIEW

**Comment by Ja4V8s28Ck** [comment]
LGTM

> 1. Just a thought, not the scope of this PR. Now that we use `generateUsername` utility function in two places, Could we move it to `helper.ts` and write some unit tests?
> 2. While creating a [new user](packages/server/src/services/auth/createUser.ts), we are creating a hashed password even before checking the user exists. As of now we only use 10 rounds of hashing in bcrypt but incase the need arises we might have to increase the number of rounds, which can get a resource heavy.

**Comment by mittalyashu** [comment]
> Just a thought, not the scope of this PR. Now that we use generateUsername utility function in two places, Could we move it to helper.ts and write some unit tests?

I will handle the part of moving the `generateUsername` function to helper file, after [this PR](https://github.com/logchimp/logchimp/pull/1494/files).

@Vedant005 Can you add test cases in this PR?

> While creating a [new user](https://github.com/logchimp/logchimp/pull/packages/server/src/services/auth/createUser.ts), we are creating a hashed password even before checking the user exists.

That is okay, in terms of code execution. As to why, you will be able to get better context in [this PR](https://github.com/logchimp/logchimp/pull/1494/files).

> As of now we only use 10 rounds of hashing in bcrypt but incase the need arises we might have to increase the number of rounds, which can get a resource heavy.

@Ja4V8s28Ck Can you create a generic issue for this and put the "RFC" label on it?

**Comment by Ja4V8s28Ck** [comment]
> That is okay, in terms of code execution. As to why, you will be able to get better context in [this PR](https://github.com/logchimp/logchimp/pull/1494/files).  

Almost forgot this PR, that makes sense.

> @Ja4V8s28Ck Can you create a generic issue for this and put the "RFC" label on it?

Done, #1502

**Comment by Vedant005** [comment]
@mittalyashu  done with the test cases

**Comment by mittalyashu** [comment]
[This PR](https://github.com/logchimp/logchimp/pull/1511) is merged, can you rebase your PR @Vedant005

**Comment by Vedant005** [comment]
@mittalyashu  rebased

**Comment by mittalyashu** [comment]
@Ja4V8s28Ck are we good to merge this PR?

File: packages/server/tests/unit/helpers.test.ts [hunk 1]
```diff
@@ -377,3 +378,65 @@ describe("generateNanoID", () => {
     }
   });
 });
+
+describe("generateUniqueUsername", () => {
+  it("should append suffix when baseUsername < 22 chars", () => {
+    const base = "Mike";
```
Comment (Ja4V8s28Ck) [inline packages/server/tests/unit/helpers.test.ts]: It would be nice if we are using faker package to generate the base names, rather than hardcoding.

Comment (Vedant005) [reply packages/server/tests/unit/helpers.test.ts]: Done

**Comment by Ja4V8s28Ck** [comment]
LGTM
